From d374e3e4d54ecc1c2f8a6fcda0601719331456b5 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 7 Nov 2022 21:54:34 +0000
Subject: [PATCH 5/9] gnu: asdf-transform-schemas: Update to 0.3.0

* gnu/packages/astronomy.scm (python-asdf-transform-schemas): Update to 0.3.0.
  [build-system]: Use new pyproject-build-system
  [tests]: Disable tests, to prevent cycle with python-asdf package.
  [native-inputs]: Remove python-pytest and python-semantic-version.
  [propagated-inputs]: Remove python-asdf. Add python-asdf-standard,
  python-importlib-resources.
---
 gnu/packages/astronomy.scm | 23 +++++++----------------
 1 file changed, 7 insertions(+), 16 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index ca1740f852..a8c51338ac 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -1990,28 +1990,19 @@ (define-public python-asdf-standard
 (define python-asdf-transform-schemas
   (package
     (name "python-asdf-transform-schemas")
-    (version "0.2.0")
+    (version "0.3.0")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "asdf_transform_schemas" version))
        (sha256
-        (base32 "1gmzd81hw4ppsvzrc91wcbjpcw9hhv9gavllv7nyi7qjb54c837g"))))
-    (build-system python-build-system)
+        (base32 "1midgn575970p5cnsh9y6bz77fjr392b5nfxb3z0id6c49xzzwhc"))))
+    (build-system pyproject-build-system)
     (arguments
-     `(#:phases
-       (modify-phases %standard-phases
-         (replace 'check
-           (lambda* (#:key inputs outputs tests? #:allow-other-keys)
-             (when tests?
-               (add-installed-pythonpath inputs outputs)
-               (invoke "python" "-m" "pytest")))))))
-    (native-inputs
-     (list python-pytest
-           python-semantic-version
-           python-setuptools-scm))
-    (propagated-inputs
-     (list python-asdf))
+     ;; Dependencies cycle with python-asdf
+     (list #:tests? #f))
+    (native-inputs (list python-setuptools-scm))
+    (propagated-inputs (list python-asdf-standard python-importlib-resources))
     (home-page "https://github.com/asdf-format/asdf-transform-schemas")
     (synopsis "ASDF schemas for transforms")
     (description
-- 
2.38.0

